internal/presentation/loop.go

```diff
@@ -33,29 +33,33 @@ type GuiHandler func(gui *gocui.Gui, view *gocui.View) error
 type gameLoop struct {
 	boardDrawer BoardDrawer
 
-	state     *model.Board
+	board     *model.Board
 	saveBoard storage.BoardSaveFn
 
 	gui *gocui.Gui
 }
 
 func (g gameLoop) insertTile(button Button, buttonIndex int) GuiHandler {
 	return func(gui *gocui.Gui, view *gocui.View) error {
+		if g.board.State != model.GameStatePlaceTile {
+			return nil
```

Comment:
shouldn't you return an error instead?

Reply:
If I return an error here, the program will crash as any error returned in an handler function makes the program stop.

Reply:
I will add a comment to provide the reason in the code.